### PR TITLE
refactor proposition

### PR DESCRIPTION
**Pull request by sqreek**
After check https://wikia-inc.atlassian.net/browse/SONAR-16 I spend few minutes to refactor code in question.

### CODE REVIEW

**Comment by sqreek** [comment]
@gabrys

File: extensions/wikia/Recirculation/services/CakeRelatedContentService.php [hunk 1]
```diff
@@ -21,105 +25,80 @@ class CakeRelatedContentService {
...
 
-			return $items;
-		} catch (ApiException $e) {
-			$this->error("error while getting content", [
-				'code' => $e->getCode(),
-				'message' => $e->getMessage(),
-			]);
-			return [];
+		$items = [];
+		$fandomArticles = array_slice( $filteredRelatedContent->getFandomArticles(), 0, $limit - count( $items ), true );
```
Comment (sqreek) [inline extensions/wikia/Recirculation/services/CakeRelatedContentService.php:79]: Forget about negative length here, but I think it's still better concept than 
`array_map(null, ...)` and `break 2`

**Comment by gabrys** [comment]
Ping @nmonterroso. I wasn't the author of this. I don't have enough context to tell either :+1: or :-1:

**Comment by nmonterroso** [comment]
ping @Wikia/cake, thanks @gabrys for looping us in!

I appreciate trying to clean up the code, but from what I can tell this changes the output to be all the fandom content, then all the discussion content, then all the wiki articles content. But the intention is to be one fandom article, one discussion article, one wiki article, etc until we reach the limit. So this is fundamentally changing what the code does, so it shouldn't be merged.

This code was also recently refactored as part of [RECIRC-200](https://github.com/Wikia/app/pull/11437/files#diff-fc87b012a152561abdce9bd9d87bccdb), so there are some conflicts that would need to be resolved.

As far as the sonar check is concerned, I agree with @gabrys comment on the JIRA:

> I could agree wrapping a whole method in a try/catch is a bad practice. Now, is having 5 return statements is a bad practice? I think it's about what you return and where than how many statements. If we need a number, because that's how SonarQube works, set it to 10 or something like that. That is most probably bad.

As long as the code is clear, I think early returns are a good way to reduce nesting in the current scope.

**Pull request closed without merging**